include/pagmo/algorithms/gwo.hpp

```diff
@@ -0,0 +1,351 @@
+/* Copyright 2017-2018 PaGMO development team
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received copies of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#ifndef PAGMO_ALGORITHMS_GWO_HPP
+#define PAGMO_ALGORITHMS_GWO_HPP
+
+#include <vector>
+#include <limits>
+#include <random>
+#include <cmath>
+#include <boost/algorithm/clamp.hpp>
+
+#include <pagmo/algorithm.hpp>
+
+namespace pagmo
+{
+/// Grey Wolf Optimizer Algorithm
+/**
+ *
+ * \image html GreyWolf.gif "One funny bee" width=3cm
+ * Grey Wolf Optimizer is an optimization algorithm based on the leadership hierarchy and hunting mechanism of greywolves,
+ * proposed by Seyedali Mirjalilia, Seyed Mohammad Mirjalilib, Andrew Lewis in 2014.
+ *
+ * The implementation provided for PaGMO is based on the pseudo-code provided in Seyedali and Andrew (2014) - Fig. 6.
+ * pagmo::GWO is suitable for box-constrained single-objective continuous optimization.
+ *
+ * See: https://www.sciencedirect.com/science/article/pii/S0965997813001853 for the paper that introduces Grey Wolf Optimizer and the pseudo-code
+ *
+ * See: https://github.com/7ossam81/EvoloPy/blob/master/GWO.py for the Python implementation
+ *
+ */
+
+class gwo{
+    public:
+            /// Single entry of the log (gen, alpha, beta, delta)
+    typedef std::tuple<unsigned int, double, double, double> log_line_type;
+    /// The log
+    typedef std::vector<log_line_type> log_type;
+
+    /// Constructor
+    /**
+     * Constructs a Grey Wolf Optimizer
+     *
+     * @param gen number of generations.
+     * @param agent number of search agents(wolf)
```

Comment:
is m_agent including the alpha beta gamma? In other words pop.size() is m_agent+3 ?

Reply:
removed m_agent as it is the same as pop.size()